_targets/build.project.armv7a7-imx6ull

```diff
@@ -113,26 +127,47 @@ b_image_target() {
 	b_log "Creating image from $PREFIX_ROOTFS"
 
 	b_mkscript_user "${UPLOAD_SCRIPT[@]}"
-	b_mkscript_preinit
-
-	# Build production image
-	b_log "RAM Image"
-	b_kernel_image "$PREFIX_BOOT/phoenix.img"
-	b_syspage_gen "$PREFIX_BOOT/phoenix.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
 
-	b_log "FLASH Image"
-	b_mkscript_user "${PRIMARY_SCRIPT[@]}"
-	b_kernel_image "$PREFIX_BOOT/phoenix-kernel.img"
-	b_syspage_gen "$PREFIX_BOOT/phoenix-kernel.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
+	if [ "$KERNEL_PLO_BOOT" = "y" ]; then
+		b_log "Image will boot from PLO."
+
+		b_log "FLASH image"
+		b_mkscript_user "${PRIMARY_SCRIPT[@]}"
+		b_prod_image
+	else
+		# Parsed by syspagen for map and console commands.
+		b_mkscript_preinit
+
+		# Build production image
+		b_log "RAM Image"
+		b_kernel_image "$PREFIX_BOOT/phoenix.img"
+		b_syspage_gen "$PREFIX_BOOT/phoenix.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
+
+		b_log "FLASH Image"
+		# This is a dirty workaround.
+		# As plo is not used user script is used only to:
+		#    - upload executable binary file instead of elf to the board(kernelimg),
+		#    - upload required apps(app).
+		b_mkscript_user "${PRIMARY_SCRIPT[@]}"
+		b_kernel_image "$PREFIX_BOOT/phoenix-kernel.img"
+		b_syspage_gen "$PREFIX_BOOT/phoenix-kernel.img" "$PLO_SCRIPT_DIR/$NAME_USER_SCRIPT"
+	fi
 
 	b_log "Creating rootfs"
 	# FIXME: does work only on macOS, for now hackish solution based on extending PATH
 	mtd-utils/build.sh
 	PATH="$(pwd)/_build/host-generic-pc/prog/:$PATH"
-	ROOTFS="$PREFIX_BOOT/rootfs.jffs2"
+	export ROOTFS="$PREFIX_BOOT/rootfs.jffs2"
+
+	if [[ $CLEANMARKER_SIZE = 0 ]]; then
+		CLEANMARKER_ARG=-n
+	else
+		CLEANMARKER_ARG="-c $CLEANMARKER_SIZE"
+	fi
+
+	mkfs.jffs2 -U $CLEANMARKER_ARG -l -m none -e $ERASE_SIZE -s $SECTOR_SIZE -r "$PREFIX_ROOTFS"/ -o "$ROOTFS" -p
```

Comment:
not sure if `-p` was not specifically omitted previously (writing `0xFF` bytes to NAND is not a no-op - it changes ECC bytes) - please verify if it's correctly skipped by `nandwrite`

Reply:
decided to remove `-p` as the method for flashing on NOR erases the JFFS2 sectors on its own